github/repos_contents.go

```diff
@@ -106,6 +121,46 @@ func (s *RepositoriesService) GetReadme(ctx context.Context, owner, repo string,
 	return readme, resp, nil
 }
 
+// GetReadme gets the Readme file for the repository.
+//
+// GitHub API docs: https://developer.github.com/v3/repos/contents/#get-the-readme
+func (s *RepositoriesService) GetReadmeCustom(ctx context.Context, owner, repo string, opt *RepositoryContentGetOptions) (string, *Response, error) {
+	u := fmt.Sprintf("repos/%v/%v/readme", owner, repo)
+	u, err := addOptions(u, opt.Params)
+	if err != nil {
+		return "", nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return "", nil, err
+	}
+
+	switch opt.MediaType {
+	case Raw:
+		req.Header.Set("Accept", mediaTypeV3Raw)
+	case HTML:
+		req.Header.Set("Accept", mediaTypeV3HTML)
+	case Object:
+		req.Header.Set("Accept", mediaTypeV3Object)
+		repoContent, resp, err := s.GetReadme(ctx, owner, repo, opt)
+		if err != nil {
+			return "", nil, err
+		}
+		content, err := repoContent.GetContent()
+		return content, resp, err
```

Comment:
L145-L151 isn't very readable. In my opinion, it will confuse any future contributors to understand what's going on in this usual piece of code.

I'll need to give it a thought about how to improve this. Let me know if you have any suggestions.